server/aesdsocket.c

```diff
@@ -243,124 +442,142 @@ int main(int argc, char *argv[])
         syslog(LOG_ERR, "Error opening file with error %d", errno);
     }
 
-    char buffer[BUFFER_SIZE];
-    int byte_recv;
-    int packet_len;
+    ret = create_timer();
+    if(ret == -1)
+    {
+        perror("timer creation");
+    }
 
-    bool data_packet = false;
-    int file_size = 0;
+    pthread_mutex_init(&mtx, NULL);
+
+    node_t *head =NULL;
+    node_t *prev,*current;
 
-    while(1)
+    while(!execution_status)
     {
-        new_fd = accept(sockfd, (struct sockaddr *)&their_addr, &addr_size);
+        if(timeout)
+        {
+            timeout = false;
+            log_timestamp();
+        }
+        int new_fd = accept(sockfd, (struct sockaddr *)&their_addr, &addr_size);
         if( new_fd == -1 )
         {
+            if(errno == EWOULDBLOCK)
```

Comment:
This works, but think about a way you wouldn't need to continuously spin the CPU here using `poll` or similar to wait for multiple fds.